pyccel/ast/utilities.py

```diff
@@ -839,3 +840,32 @@ def is_literal_integer(expr):
     """
     return isinstance(expr, (int, LiteralInteger)) or \
         isinstance(expr, PyccelUnarySub) and isinstance(expr.args[0], (int, LiteralInteger))
+
+#==============================================================================
+def flatten_tuple_var(expr, scope):
```

Comment:
Do you intend to use this function with nested inhomogeneous tuples?

Reply:
Yes that is the aim. It will become more important when I fix #337

Reply:
That's what I thought, but such a mapping is lossy? shouldn't the strides and shape be computed here as well? or is it taken care of in a separate function?

Reply:
I'm not sure what you mean. `BindCArrayType` inherits from `InhomogeneousTupleType` so the elements of this tuple are the shape and stride variables. You lose the information about how they are linked to one another but this function is only used when that information is not required. Eg to print `a = get_array` we call `flatten_tuple_var` on `a` to get `(a_ptr, a_shape0, a_shape1)` (no strides because this is a return type) so we can print
```
get_array(&a_ptr, &a_shape0, &a_shape1)
```

The way they are linked is still available elsewhere in the code because the result is not saved in a normal `InhomogeneousTupleType` but rather in a `BindCArrayType`.

Reply:
If a function returned `(1, my_array)` then this method would see `(1, (my_array_ptr, my_array_shape))` and would return `(1, my_array_ptr, my_array_shape)`

Reply:
Got it. Thank you!